### DISCUSSION TITLE
Question: metrics initialisation in Prometheus-exporter

### DISCUSSION THREAD

**Discussion by lswith** [metrics-rs/metrics#376] [Q&A] [answered]
Hi,
If I have registered a `counter`, does it get set to 0 until incremented? Or does it simply not get set to 0?

The reason I ask, is that a lot of our prometheus dashboards/alerts expect that there be a value, even if it's 0 as this indicates that the metric is still being emitted (and not an issue with the scraping of the metric etc).

I'm hoping I can also remove the idleTimeout so that even stale metrics get reported consistently.

**Comment by tobz** [comment, marked as answer]
> If I have registered a `counter`, does it get set to 0 until incremented? Or does it simply not get set to 0?

Counters start life at a zero value, yes.

> I'm hoping I can also remove the idleTimeout so that even stale metrics get reported consistently.

There's no idle timeout by default, so just... don't set it. :)
